src/test/java/seedu/address/logic/commands/AddVisitCommandTest.java

```diff
@@ -4,43 +4,96 @@
 import seedu.address.model.VisitBook;
 import seedu.address.model.visit.Visit;
 import seedu.address.testutil.VisitBuilder;
 
 public class AddVisitCommandTest {
+
     @Test
-    public void constructor_nullLocation_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> new AddVisitCommand(null));
+    public void constructor_nullInput_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new AddVisitCommand(null, null, null));
+        assertThrows(NullPointerException.class, () -> new AddVisitCommand(DEFAULT_PERSON_INDEX,
+                DEFAULT_LOCATION_INDEX, null));
+        assertThrows(NullPointerException.class, () -> new AddVisitCommand(DEFAULT_PERSON_INDEX,
+                null, DEFAULT_DATE));
+        assertThrows(NullPointerException.class, () -> new AddVisitCommand(null,
+                DEFAULT_LOCATION_INDEX, DEFAULT_DATE));
     }
 
     @Test
-    public void execute_visitAcceptedByModel_addSuccessful() throws Exception {
+    public void execute_unfilteredList_addSuccessful() {
         ModelStubAcceptingVisitAdded modelStub =
                 new AddVisitCommandTest.ModelStubAcceptingVisitAdded();
         Visit validVisit = new VisitBuilder().build();
 
-        CommandResult commandResult = new AddVisitCommand(validVisit).execute(modelStub);
+        try {
+            CommandResult commandResult = new AddVisitCommand(DEFAULT_PERSON_INDEX, DEFAULT_LOCATION_INDEX,
+                    DEFAULT_DATE).execute(modelStub);
+
+            assertEquals(String.format(AddVisitCommand.MESSAGE_SUCCESS, validVisit),
+                    commandResult.getFeedbackToUser());
+            assertEquals(Arrays.asList(validVisit), modelStub.visitsAdded);
+        } catch (CommandException e) {
+            assert false : "Command Exception not expected.";
+        }
+    }
 
-        assertEquals(String.format(AddVisitCommand.MESSAGE_SUCCESS, validVisit),
-                commandResult.getFeedbackToUser());
-        assertEquals(Arrays.asList(validVisit), modelStub.visitsAdded);
+    @Test
+    public void execute_filteredList_success() {
+        Model model = new ModelManager(getTypicalAddressBook(), getTypicalLocationBook(),
+                new UserPrefs(), new VisitBook());
+
+        showLocationAtIndex(model, INDEX_THIRD);
+        showPersonAtIndex(model, INDEX_FOURTH);
+
+        Index personId = model.getFilteredPersonList().get(0).getId();
+        Index locationId = model.getFilteredLocationList().get(0).getId();
+        Visit validVisit = new VisitBuilder().withDate(DEFAULT_DATE_STRING)
+                .withLocationId(Integer.toString(locationId.getOneBased()))
+                .withPersonId(Integer.toString(personId.getOneBased()))
+                .build();
+
+        try {
+            new AddVisitCommand(INDEX_FIRST, INDEX_FIRST, DEFAULT_DATE).execute(model);
+            assertEquals(model.getFilteredVisitList().get(0), validVisit);
```

Comment:
Just a thought, shouldn't the validVisit be created based on the hard coded locationId INDEX_THIRD and personId INDEX_FOURTH? Right now valid visit still references the filtered list. If there's an error in the translation of ids between filtered list and the original lists, the test would still pass as both the AddVisitCommand and validVisit would have made the same error.

Reply:
This copy of valid visit is hard coded, by using the visit builder class. The hard coded values are retrieved using the getOneBased method. Valid visit has no direct reference to filtered visit at all.

The gist of this test is that my model's filtered lists for person and location currently store only 1 location with id 3 and 1 person with id 4. This is done using the showLocationAtIndex, showPersonAtIndex methods above. Im giving the model input using index 1 and index 1 for person and location, because the list user sees contains only 1 element. After processing however, I am expecting the visit to point to ids 3 and 4 respectively. This is done by comparing the final visit from the visit list with the expected visit created using valid visit.

Reply:
I see, no issue then!